packages/myst-cli/src/init/jupyter-book/syntax.ts

```diff
@@ -0,0 +1,230 @@
+    documentPaths = await glob('**/*.{md,ipynb}');
+  }
+  // Update all documents
+  await Promise.all(documentPaths.map((path) => upgradeDocument(session, path)));
+}
+
+/**
+ * In-place upgrade Sphinx-style glossaries into MyST definition-list glossaries
+ * in a single document
+ *
+ * @param session - session with logging
+ * @param path - path to document
+ */
+async function upgradeDocument(session: ISession, path: string) {
+  const backupFilePath = makeBackupName(path);
+
+  // Ensure that we havent' already done this once
+  if (await fsExists(backupFilePath)) {
+    return;
+  }
+
+  const { ext } = parse(path);
+  switch (ext) {
+    case '.md':
+      {
+        // Upgrade entire Markdown document in one pass
+        const data = (await fs.readFile(path)).toString();
+        const maybeNewLines = await upgradeContent(data.split(SPLIT_PATTERN));
+        if (maybeNewLines !== undefined) {
+          // Backup existing file
+          await fs.rename(path, backupFilePath);
+
+          // Write modified result
+          session.log.info(chalk.dim(`Backed up original version of ${path} to ${backupFilePath}`));
+          await fs.writeFile(path, maybeNewLines.join('\n'));
+        }
+      }
+      break;
+    case '.ipynb': {
+      // Upgrade each cell of the notebook
+      const data = (await fs.readFile(path)).toString();
+      const notebook = JSON.parse(data) as INotebookContent;
+      const cellDidUpgrade = await Promise.all(
+        notebook.cells
+          .filter((cell) => cell.cell_type === 'markdown')
+          .map(async (cell) => {
+            // Try and upgrade the cell
+            const maybeNewLines = await upgradeContent(
+              (cell.source as string[])
+                // Strip newlines
+                .map((line) => line.replace('\n', '')),
+            );
+            // Did we compute new state?
+            if (maybeNewLines !== undefined) {
+              // Write to cell and indicate modification
+              cell.source = maybeNewLines.map((line) => `${line}\n`);
+              return true;
+            } else {
+              return false;
+            }
+          }),
+      );
+
+      // Do we need to update the notebook?
+      if (cellDidUpgrade.some((x) => x)) {
+        // Backup existing file
+        await fs.rename(path, backupFilePath);
+
+        // Write modified result
+        const newData = JSON.stringify(notebook);
+        session.log.info(chalk.dim(`Backed up original version of ${path} to ${backupFilePath}`));
+        await fs.writeFile(path, newData);
+      }
+    }
+  }
+}
+
+async function upgradeContent(documentLines: string[]): Promise<string[] | undefined> {
+  let didUpgrade = false;
+
+  for (const transform of [upgradeGlossary]) {
+    const nextLines = await transform(documentLines);
+    didUpgrade = didUpgrade || nextLines !== undefined;
+    documentLines = nextLines ?? documentLines;
+  }
+
+  return didUpgrade ? documentLines : undefined;
```

Comment:
Should we do `:::{Note}` here? Can kick that down the line.

Reply:
Let's kick that down the road?